grails-app/services/org/bbop/apollo/RequestHandlingService.groovy

```diff
@@ -1270,8 +1270,12 @@ class RequestHandlingService {
 
         for (int i = 0; i < features.length(); ++i) {
             JSONObject jsonFeature = features.getJSONObject(i);
-            int fmin = jsonFeature.get(FeatureStringEnum.LOCATION.value).fmin
-            int fmax = jsonFeature.get(FeatureStringEnum.LOCATION.value).fmax
+            Integer fmin = jsonFeature.get(FeatureStringEnum.LOCATION.value)?.fmin
+            Integer fmax = jsonFeature.get(FeatureStringEnum.LOCATION.value)?.fmax
+            if(!fmax){
+                fmax = fmin
```

Comment:
This is an incorrect assumption.

`fmax = fmin` is valid only for insertion.
For deletion and substitution, `fmax = fmin + 1`

Shouldn't it be the responsibility of the client to provide proper `fmin` and `fmax`?
Inferring the `fmax` can only lead to problems. Instead, we should throw an AnnotationException when the incoming JSON doesn't have either the `fmin` or the `fmax`.